Source/WebCore/loader/cache/CachedFont.cpp

```diff
@@ -64,9 +67,38 @@ void CachedFont::didAddClient(CachedResourceClient& client)
         downcast<CachedFontClient>(client).fontLoaded(*this);
 }
 
+static String hashForFontData(const uint8_t* data, size_t size)
+{
+    auto cryptoDigest = PAL::CryptoDigest::create(PAL::CryptoDigest::Algorithm::SHA_256);
+    cryptoDigest->addBytes(data, size);
+    auto digest = cryptoDigest->computeHash();
+    return makeString("sha256-", base64Encoded(digest.data(), digest.size()));
+}
+
+static bool isAllowedFont(const String& fontSha256)
+{
+    // proof-of-concept for a .woff open-sans file hosted at https://fonts.gstatic.com/s/opensans/v34/memSYaGs126MiZpBA-UvWbX2vVnXBbObj2OVZyOOSr4dVJWUgsjZ0B4kaVQUwaEQbjB_mQ.woff
+    const String sha256ForAllowedResource { "sha256-+rqTf0QR0w8EDckgjKTUoZ4Gdl0eKoDc+GSrLlpZ+ks="_s };
+    // std::unordered_set<String> allowedShas { sha256ForAllowedResource };
+    Vector<String> allowedShas { sha256ForAllowedResource };
```

Comment:
Consider using `MemoryCompactLookupOnlyRobinHoodHashSet` once you have more than one entry.